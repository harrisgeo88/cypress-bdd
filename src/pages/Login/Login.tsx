import React, { useState } from 'react'
import { InputText } from '../../components/InputText'
import { Layout, Wrapper } from '../../components/Layout'
import { Button } from '../../components/Button'
import { submitLogin } from './Login.utils'

export const Login = () => {
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')
  const [loggedIn, setLoggedIn] = useState(false)

  const handleClick = () => {
    if (submitLogin(username, password)) {
      setLoggedIn(true)
    }
  }

  // const handleClick = () => {
  //   if (username === 'harrisgeo88' && password === 'pass') {
  //     setLoggedIn(true)
  //   }
  // }

  return (
    <Layout>
      <Wrapper>
        {!loggedIn && (
          <>
            <InputText
              placeholder='Username'
              data-testid='username'
              value={username}
              onChange={value => setUsername(value)}
            />
            <InputText
              type='password'
              data-testid='password'
              placeholder='Password'
              value={password}
              onChange={value => setPassword(value)}
            />
            <Button data-testid='submit' onClick={handleClick}>
              Submit
            </Button>
          </>
        )}
        {loggedIn && (
          <>
            <div data-testid='loggedin'>You are logged in</div>
          </>
        )}
      </Wrapper>
      <div></div>
    </Layout>
  )
}
