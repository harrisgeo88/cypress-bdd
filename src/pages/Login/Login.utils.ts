export const submitLogin = (username: string, password: string): Boolean => {
  if (username === 'master' && password === 'pass') {
    return true
  }
  if (username === 'harrisgeo88' && password === 'pass') {
    return true
  }

  return false
}
