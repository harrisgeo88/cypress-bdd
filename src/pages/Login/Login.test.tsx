import { Login } from './Login'
import { render } from '@testing-library/react'

describe('given an `Login` component', () => {
  describe('when it is rendered', () => {
    it('should match snapshot', () => {
      const { container } = render(<Login />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
