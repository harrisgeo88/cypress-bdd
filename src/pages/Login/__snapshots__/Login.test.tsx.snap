// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given an \`Login\` component when it is rendered should match snapshot 1`] = `
<div
  class="Layoutstyles__Layout-sc-8y4jny-0 cDaGbz"
>
  <div
    class="Layoutstyles__Wrapper-sc-8y4jny-1 cwLOSA"
  >
    <div
      class="InputTextcomponents__InputTextWrapper-sc-11nm3a5-0 ehojjc"
    >
      <input
        class="InputTextcomponents__Input-sc-11nm3a5-1 lhFXMG"
        data-testid="username"
        placeholder="Username"
        value=""
      />
    </div>
    <div
      class="InputTextcomponents__InputTextWrapper-sc-11nm3a5-0 ehojjc"
    >
      <input
        class="InputTextcomponents__Input-sc-11nm3a5-1 lhFXMG"
        data-testid="password"
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      class="Buttonstyles__ButtonWrapper-sc-11g5hpj-0 gvIbjm"
      data-testid="submit"
      type="button"
    >
      Submit
    </button>
  </div>
  <div />
</div>
`;
